$NetBSD: patch-tools_debugserver_source_CMakeLists.txt,v 1.1 2026/03/25 22:56:25 wiz Exp $

On Darwin, fix build when CMAKE_OSX_SYSROOT is not defined or is empty.

--- tools/debugserver/source/CMakeLists.txt.orig	2026-02-24 20:06:40.978711491 +0000
+++ tools/debugserver/source/CMakeLists.txt
@@ -176,10 +176,17 @@ if (DEFINED MIG_ARCHS)
 endif()
 separate_arguments(MIG_ARCH_FLAGS_SEPARTED NATIVE_COMMAND "${MIG_ARCH_FLAGS}")
 
+if(DEFINED CMAKE_OSX_SYSROOT AND CMAKE_OSX_SYSROOT)
 add_custom_command(OUTPUT ${generated_mach_interfaces}
   VERBATIM COMMAND mig ${MIG_ARCH_FLAGS_SEPARTED} -isysroot ${CMAKE_OSX_SYSROOT} ${CMAKE_CURRENT_SOURCE_DIR}/MacOSX/dbgnub-mig.defs
   DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/MacOSX/dbgnub-mig.defs
   )
+else()
+add_custom_command(OUTPUT ${generated_mach_interfaces}
+  VERBATIM COMMAND mig ${MIG_ARCH_FLAGS_SEPARTED} ${CMAKE_CURRENT_SOURCE_DIR}/MacOSX/dbgnub-mig.defs
+  DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/MacOSX/dbgnub-mig.defs
+  )
+endif()
 
 set(DEBUGSERVER_VERS_GENERATED_FILE ${CMAKE_CURRENT_BINARY_DIR}/debugserver_vers.c)
 configure_file(debugserver_vers.c.in
